sherpa/data.py

```diff
@@ -52,6 +52,24 @@ def _check(array):
     return array
 
 
+def _check_nomask(array):
```

Comment:
When I see a message like "Dropping mask attribute" I think I'd expect for the attribute to be removed in some way. I'm not sure that's the behavior here - the .mask attribute may stay but it is just ignored. Or am I missing something?

I'm not 100% convinced that `del array.mask` would be a sensible thing here, but I wanted to understand the warning message.

Reply:
I don't want to `del` anything, because we don't want to change the original array that the user had. I've reworded the messages, you'll see that on my next push.